docs/index.md

```diff
@@ -4,396 +4,182 @@
-
-Generative AI has a ton of variation in results, so experimenting is super
-important. The local Genkit Developer UI lets you interact with essential AI
-components like models and retrievers, as well as manually test your end-to-end
-flows including all the custom code you wrote.
-
-<video autoplay muted loop playsinline>
-  <source src="resources/devui-run.webm" type="video/webm" />
-</video>
-
-### 7. Inspect traces
-
-Debugging complex, multi-step workflows with AI can be challenging due to
-randomness and hidden processes. Genkit provides a trace inspector in the
-developer UI that lets you inspect traces for each model call and step in your
-flow. It can view traces from production and even render images!
-
-<video autoplay muted loop playsinline>
-  <source src="resources/devui-trace.webm" type="video/webm" />
-</video>
-
-### 8. Open & extensible
-
-The AI ecosystem is growing faster than any one team can keep up with. Genkit
-has an open plugin model that provides pre-built integrations with new models,
-retrievers, and more. While the Genkit team maintains a small set of official
-plugins, anyone can feel free to publish their own Genkit plugins to NPM.
-
-Can't find a plugin for a particular integration you want? No problem. Genkit's
-abstractions are flexible and make it easy to build custom components that
-integrate into the framework, like this custom Firestore retriever:
-
-```js
-import { embed } from '@genkit-ai/ai/embedder';
-import { Document, defineRetriever } from '@genkit-ai/ai/retriever';
-import { textEmbeddingGecko } from '@genkit-ai/vertexai';
-import {
-  FieldValue,
-  VectorQuery,
-  VectorQuerySnapshot,
-} from '@google-cloud/firestore';
-import { Firestore } from 'firebase-admin/firestore';
-import * as z from 'zod';
-import { augmentedPrompt } from './prompt';
-
-const QueryOptions = z.object({
-  k: z.number().optional(),
-});
-
-const firestoreArtifactsRetriever = defineRetriever(
-  {
-    name: 'firestore/artifacts',
-    configSchema: QueryOptions,
-  },
-  async (input, options) => {
-    const embedding = await embed({
-      embedder: textEmbeddingGecko,
-      content: input,
-    });
-
-    const db = new Firestore();
-    const coll = db.collection('vectors' /* your collection name */);
-
-    const vectorQuery: VectorQuery = coll.findNearest(
-      'embedding' /* the name of the field that contains the vector */,
-      FieldValue.vector(embedding),
-      {
-        limit: options.k ?? 3,
-        distanceMeasure: 'COSINE',
+      prompt: 'Recommend a restaurant in San Francisco for a romantic dinner.',
+      // Specify output structure using Zod schema
+      output: {
+          schema: z.object({
+              restaurantName: z.string(),
+              cuisine: z.string(),
+              address: z.string(),
+              rating: z.number().min(1).max(5),
+              priceTier: z.enum(['$', '$$', '$$$', '$$$$']),
+              highlights: z.array(z.string())
+          })
```

Comment:
I don't like this example because it's asking the LLM to hallucinate a potentially bogus result. Prefer a creative task like "Generate a creature for a fantasy RPG" or something else that doesn't require factual accuracy of the model.

Reply:
Fixed